### ISSUE TITLE
[minor] When switching languages, the project name isn't changed

### ISSUE THREAD

**Issue by cycomachead** [jmoenig/Snap--Build-Your-Own-Blocks#972]
This probably only occurs for people testing snap, and I think all that's missing is a call to `localize()` in the right spot...

Basically, after switching languages 1, the project's name (if untitled) gets suck at the first language you switch to.

Repro:
* Open a new project
* Switch to a language
* Project name will change
* Switch to a new language

**Comment by jmoenig** [comment]
Yes, that's absolutely intentional! Of *course* we don't translate any *names*, think about it! The same applies for sprite names, the stage, custom block specs, costume names, sound names. I've seen some people point this out, but I never replied because to me that's totally obvious :-)

**Comment by cycomachead** [comment]
No, no, I'm not saying we should translate _all_ names, this bug specifically applies to the "Untitled" case, which has calls to `localize()` in some, but not all cases.

English, project is "untitled" → German, now I have a german name → English, my project still has a german name. 

Perhaps I should have added "very" minor to title, but it was just an odd occurance when I was testing.

**Comment by jmoenig** [comment]
Yes, it does have a call to ```localize()``` but only if the user creates a *new* project. Again, this is absolutely what I want it to be. I guess to *really* address this problem we could check whether a user has made *any* edits to a *new* project *before* switching languages. In that case (and in that case alone) it might be okay to translate the default project name.

**Comment by cycomachead** [comment]
> Yes, it does have a call to localize() but only if the user creates a new project.

What I'm saying is that it doesn't really make sense to localize the name Untitled when you change from english to another language, but not when changing the language in anything other than english to anything else (including English). 

Actually though, it's not about missing a call, or it could be, but the real issue is in English a new project has the name "" but _displays_ "untitled", whereas in every other language, a new project has the actual name `localize('untitled')`. This is different from what I initally expected! 

But really, the naming is not a big deal.  However, what is somewhat odd, in every other language there's no dialog when trying to save a project for the first time, because the name is set to something already. I now have a lot of "Untitled" projects in a variety of languages. ;)

**Comment by jmoenig** [comment]
> in every other language there's no dialog when trying to save a project for the first time

No, that's not right. Whenever you crate a new project in *any* language, saving (or exporting) it for the first time prompts you for a name.

See, what happens when you switch languages or zoom blocks or change layouts is this: Snap *serializes* the current project, applies the change, and then *deserializes* (opens) the buffered project again. In *these* cases, if a project doesn't yet have a name, Snap does *not* ask you for a name, but instead applies the "untitled" string to the project, so you can switch back and forth, zoom blocks etc. without getting nagged by dialog boxes. *That's* the issue, if you will.

**Comment by cycomachead** [comment]
> No, that's not right. Whenever you create a new project in any language, saving (or exporting) it for the first time prompts you for a name.

When I go to s.b.e/run with my language set to German, and hit Cmd-S, a dialog asking me to enter a name does _not_ popup. When I look at the ide's properties in the inspector, `projectName` already has a value.

**Comment by jmoenig** [comment]
Oh, right. When you first load Snap in another language it actually doesn't *load* it in that other language, but *switches* languages afterwards. But as soon as you make a *new* project Snap asks you for a project name.
